Bugfixes: Erratic mouselook; Terrain elevator nonstop sound; Cutscene audio; Mouselook held obj; Shift key sprint; Cutscene window resize; Gamma slider

donnierussellii opened the pull request:
Clicking captured mouse cursor on window resize border (why is this allowed by SDL?) in windowed, non-mouselook mode, then switching to mouselook mode causes mouselook to behave erratically.

When capturing mouse, now also disable window border drag resize. Also added hack to actually keep captured mouse cursor inside client area.

**Edit1:** Added a fix for mouselook drift when clicking. Mouselook now only checks actual mouse move events.

**Edit2:** Added fix for terrain elevator sound not stopping when reversing an already moving terr elevator.

**Edit3:** Don't play cutscene audio if sound is turned off in options. Before this fix, cutscene audio did not stop when skipping cutscene. Also set data member of snd_digi_parms to NULL; otherwise uninitialized.

**Edit4:** Fix for mouselook toggle making held object change to crosshair, while obj still held.

**Edit5:** Hack to allow pressing shift after move key. For example, while holding 'w', repeatedly press and release shift to alternately run and walk.

**Edit6:** Hide mouse cursor during cutscenes. The transparent area around the cursor is not drawing properly anyway.

**Edit7:** Don't change screen size during cutscenes or after exiting full map display, and make changing screen size smarter. Uncapture mouse during cutscenes.

**Edit8:** Fixed gamma slider in options and gamma update in game loop.

**Edit9:** Stop all sound channels when pausing.

**Edit10:** Compensate for difference in brightness between fullscreen and windowed mode. May be platform-specific--tested on Windows.

Interrupt commented on src/GameSrc/setup.c:
```diff
@@ -1165,7 +1165,7 @@ void pause_for_key(ulong wait_time) {
-void splash_draw() {
+void splash_draw(bool show_splash) {
     int pal_file;
 
```
We could probably just return early here, would make the logic clearer to follow than all of the added if blocks.

Interrupt replied:
Might not have pushed that change, not seeing it in the commit log. Want to double check before I merge?

donnierussellii commented:
You're right--I'll make the change.

On Sat, Sep 1, 2018 at 11:54 AM Chad Cuddigan <notifications@github.com>
wrote:

> *@Interrupt* commented on this pull request.
> ------------------------------
>
> In src/GameSrc/setup.c
> <https://github.com/Interrupt/systemshock/pull/165#discussion_r214517090>:
>
> > @@ -1165,7 +1165,7 @@ void pause_for_key(ulong wait_time) {
>      waiting_for_key = false;
>  }
>
> -void splash_draw() {
> +void splash_draw(bool show_splash) {
>      int pal_file;
>
>
> We could probably just return early here, would make the logic clearer to
> follow than all of the added if blocks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Interrupt/systemshock/pull/165#pullrequestreview-151596933>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKfDAny6OmApnC6BJHGvEHAQf97C9QRIks5uWq27gaJpZM4WUksr>
> .
>

Interrupt commented:
Bug in this: alt+enter no longer goes fullscreen on OSX, it just maximizes the window. Mouse look in that maximized window mode also only moves the view vertically, it's not possible to move the view back toward the ground.

donnierussellii commented:
I changed the fullscreen code back to use SDL_SetWindowFullscreen. I also added a hack to ensure the captured mouse cursor stays inside the client area.

I need some feedback on this: On my Windows system, fullscreen mode displays brighter than windowed. This is not related to the gamma setting--that changes the palette. When the gamma is set to max, fullscreen mode is extremely bright and windowed is merely bright.

So I set the brightness to 0.5 when fullscreen, and 1.0 when windowed to compensate. Is this behavior the same on other platforms (or other Windows systems)? I can put in an ifdef for Windows only if needed.

Interrupt commented:
That sounds like hardware gamma, not the in-game palette based gamma. You can mess with that via SDL using the `SDL_SetWindowBrightness` if you want to try some experiments.
